src/Compilers/Core/Portable/SourceGeneration/Nodes/InputNode.cs

```diff
@@ -44,8 +45,25 @@ public NodeStateTable<T> UpdateStateTable(DriverStateTable.Builder graphState, N
             var inputItems = _getInput(graphState);
             TimeSpan elapsedTime = stopwatch.Elapsed;
 
+            HashSet<T> itemsSet;
+            PooledHashSet<T>? pooledItemsSet;
```

Comment:
Rather than this dual scheme, I think it will be much easier to just do `(itemsSet as PooledHashSet<T>)?.Free()` at the end.

Reply:
Yeah, that is nicer